motionplan/planManager.go

```diff
@@ -77,6 +78,12 @@ func (pm *planManager) PlanSingleWaypoint(ctx context.Context,
 	var goals []spatialmath.Pose
 	var opts []*plannerOptions
 
+	// Viability check; ensure that the waypoint is not impossible to reach
+	_, err = pm.getSolutions(ctx, goalPos, seed)
```

Comment:
isn't this repeating work?

Reply:
Thinking about this more... You want to check ahead of time if you're going to be wasting time looking for a solution that isn't there.  Why don't we get rid of this and do the search back to front?  If we get a fail on the original leg of the plan we will know it fails

Reply:
This does repeat work, you're correct; it's unfortunately pretty non-trivial to actually carry the solutions forwards to where they would be needed, since it's not the ultimate solver doing the solving here. We'd need to update the motionPlanner interface to add a list of IK solutions to `plan()`, which is doable, but feels like more than we might want to do right now. The impact of a single extra IK solve here (rather than, say, one per sub-waypoint or multiple per motion plan) is pretty minimal.

We can't solve back-to-front because we want to seed the IK solver with the actual starting joint positions for the movement; otherwise we can't filter by goodness of solution, and also bias the solver against good solutions.

Reply:
I agree that the repeating work isn't ideal but there's not really an easy way out. Either we commit to a larger arch change, or do this, which is a ~20ms hit per move request.

Reply:
Actually, looking at the failed tests, the current impl is wrong and it needs to use the actual planner so that it gets the intended metric, which might be less strict than what's being used here. So the update to `plan()` above makes sense

Reply:
I don't follow your last point, what tests are you referring to?

Reply:
https://github.com/viamrobotics/rdk/actions/runs/4248035943/jobs/7386737426

Reply:
Ok I'm good to add this in for now but I have the feeling that repeating this work means we've designed something wrong somewhere along the line, so I think this would be good to call out in a TODO or ticket

Reply:
The solution I've proposed above eliminates all repeated work